### ISSUE TITLE
aten::mean silently ignores dtype and exports the wrong output type

### ISSUE THREAD

**Issue by promiseeuler** [microsoft/onnxscript#3008]
## Summary

While checking TorchLib operator signatures after reproducing #3002, I found that the no-dimension `aten::mean` implementation does not accept its schema's `dtype` argument. Unlike the random-operator failure, this export does not raise: it succeeds while silently producing a `FLOAT` ONNX output for `torch.mean(..., dtype=torch.float64)`.

## Environment

- ONNXScript current source: `0.7.2.dev20260822`
- PyTorch `2.10.0`
- ONNX `1.22.0`
- Python `3.14.2`
- macOS, Apple Silicon

## Steps to reproduce

```python
import onnx
import torch

class Model(torch.nn.Module):
    def forward(self, x):
        return torch.mean(x, dtype=torch.float64)

x = torch.tensor([[1e8, 1.0, -1e8]], dtype=torch.float32)
expected = Model()(x)
torch.onnx.export(Model(), (x,), "mean_dtype.onnx", dynamo=True)

model = onnx.load("mean_dtype.onnx")
actual_type = onnx.TensorProto.DataType.Name(
    model.graph.output[0].type.tensor_type.elem_type
)
print(expected.dtype, expected.item())
print(actual_type)
```

## Expected behavior

```text
torch.float64 0.3333333333333333
DOUBLE
```

The cast must happen before the reduction so accumulation uses the requested dtype.

## Actual behavior

```text
torch.float64 0.3333333333333333
FLOAT
```

The exported graph contains only `ReduceMean -> Squeeze`; no cast is emitted.

## Possible cause

`onnxscript/function_libs/torch_lib/ops/core.py` registers:

```python
@torch_op("aten::mean")
def aten_mean(self: TReal) -> TReal:
    """mean(Tensor self, *, ScalarType? dtype=None) -> Tensor"""
```

The docstring and PyTorch schema include `dtype`, but the function signature and body omit it. The exporter therefore lowers the operation without the requested type instead of rejecting the call.

## Proposed solution

Accept `dtype: int = -1` and cast `self` before `ReduceMean` when it is supplied. I tested this temporary local change:

```python
def aten_mean(self: TReal, dtype: int = -1) -> TReal:
    if dtype != -1:
        self = op.Cast(self, to=dtype)
    result = op.ReduceMean(self)
    return op.Squeeze(result)
```

The same reproduction then exported `Cast -> ReduceMean -> Squeeze` and declared the output as `DOUBLE`.

A regression test should assert both output type and a precision-sensitive value such as `[1e8, 1, -1e8]`, so a cast-after-reduction implementation cannot pass accidentally.

**Comment by Om-singhaI** [comment]
I have a fix for this with a regression test in progress (dtype on the no dim aten::mean path, cast before the reduction as you describe) and will open a PR shortly.
